src/libraries/System.Memory/tests/MemoryMarshal/GetRawArrayData.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.SpanTests
+{
+    public static partial class MemoryMarshalTests
+    {
+        [Fact]
+        public static void GetRawArrayData_NullInput_ThrowsNullRef()
+        {
+            Assert.Throws<NullReferenceException>(() => MemoryMarshal.GetRawArrayData((object[])null));
+        }
+
+        [Fact]
+        public static void GetRawArrayData_NonEmptyInput_ReturnsRefToFirstElement()
+        {
+            int[] theArray = new int[] { 10, 20, 30 };
+            Assert.True(Unsafe.AreSame(ref theArray[0], ref MemoryMarshal.GetRawArrayData(theArray)));
+        }
+
+        [Fact]
+        public static unsafe void GetRawArrayData_EmptyInput_ReturnsRefToWhereFirstElementWouldBe()
+        {
+            int[] theArray = new int[0];
```

Comment:
You mentioned above in the API doc:
> Such a reference may be used for pinning but must never be dereferenced.

What will happen if you did deference it?

Reply:
The behavior is undefined. It will most likely return a nonsensical value, but in theory it could also AV the process. Not something we can test for unfortunately.